github/Repository.py

```diff
@@ -1273,23 +1274,27 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert isinstance(generate_release_notes, bool), generate_release_notes
+        assert isinstance(name, str) or generate_release_notes and (name is None or is_optional(name, str)), name
+        assert (
+            isinstance(message, str) or generate_release_notes and (message is None or is_optional(message, str))
+        ), message
```

Comment:
This project doesn't use `None` for unset optional arguments but `NotSet`. Staying with the convention:
```suggestion
        assert isinstance(name, str) or generate_release_notes and is_optional(name, str), name
        assert (
            isinstance(message, str) or generate_release_notes and is_optional(message, str)
        ), message
```